src/config/config.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <fstream>
+
+#include <utils/json.hpp>
+
+#include "config.hpp"
+#include "log.hpp"
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static aos::common::utils::Duration GetDuration(
+    const aos::common::utils::CaseInsensitiveObjectWrapper& object, const std::string& key)
+{
+    auto value = object.GetValue<std::string>(key);
+
+    if (value.empty()) {
+        return {};
+    }
+
+    auto ret = aos::common::utils::ParseDuration(value);
+
+    if (!ret.mError.IsNone()) {
+        throw std::runtime_error("Failed to parse " + key + ": " + std::string(ret.mError.Message()));
+    }
+
+    return ret.mValue;
+}
+
+static Downloader ParseDownloader(const aos::common::utils::CaseInsensitiveObjectWrapper& object)
+{
+    return Downloader {
+        object.GetValue<std::string>("DownloadDir"),
+        object.GetValue<int>("MaxConcurrentDownloads"),
+        GetDuration(object, "RetryDelay"),
+        GetDuration(object, "MaxRetryDelay"),
+    };
+}
+
+static VChanConfig ParseVChanConfig(const aos::common::utils::CaseInsensitiveObjectWrapper& object)
+{
+    return VChanConfig {
+        object.GetValue<int>("Domain"),
+        object.GetValue<std::string>("XSOpenRXPath"),
+        object.GetValue<std::string>("XSOpenTXPath"),
+        object.GetValue<std::string>("XSSecureRXPath"),
+        object.GetValue<std::string>("XSSecureTXPath"),
+        object.GetValue<std::string>("CertStorage"),
+    };
+}
+
+static IAMConfig ParseIAMConfig(const aos::common::utils::CaseInsensitiveObjectWrapper& object)
+{
+    return IAMConfig {
+        object.GetValue<std::string>("IAMServerURL"),
+        object.GetValue<std::string>("CertStorage"),
+    };
+}
+
+/***********************************************************************************************************************
+ * Public functions
+ **********************************************************************************************************************/
+
+aos::RetWithError<Config> ParseConfig(const std::string& filename)
+{
+    LOG_DBG() << "Parsing config file: " << filename.c_str();
+
+    std::ifstream file(filename);
+
+    if (!file.is_open()) {
+        return {Config {}, aos::ErrorEnum::eNotFound};
+    }
+
+    auto result = aos::common::utils::ParseJson(file);
+    if (!result.mError.IsNone()) {
+        return {Config {}, result.mError};
+    }
+
+    Config config {};
+
+    try {
+        aos::common::utils::CaseInsensitiveObjectWrapper object(result.mValue.extract<Poco::JSON::Object::Ptr>());
+
+        config.mWorkingDir         = object.GetValue<std::string>("WorkingDir");
```

Comment:
Please rename these tag names.
Previously we agreed to use lower came case style
```suggestion
        config.mWorkingDir         = object.GetValue<std::string>("workingDir");
```